### PR TITLE
SAA-221 Create Activities Management API location endpoints

### PR DESCRIPTION
**Pull request by markford-moj**
New endpoint - cell locations by group name.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/LocationController.kt [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        description = "Requested resource not found",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      )
+    ]
+  )
+  fun getLocationGroups(
+    @PathVariable("prisonCode") prisonCode: String,
+    @RequestParam(value = "groupName", required = true) groupName: String,
+  ): List<Location>? {
+    return locationService.getCellLocationsForGroup(prisonCode, groupName)
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/LocationController.kt:68]: Just a style thingy, with one liner methods like this you can simply do:

```
fun getLocationGroups(
    @PathVariable("prisonCode") prisonCode: String,
    @RequestParam(value = "groupName", required = true) groupName: String,
   ) = locationService.getCellLocationsForGroup(prisonCode, groupName)
```

File: src/test/resources/__files/prisonapi/location-404.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
   "status": 404,
-  "userMessage": "(user message)Location groups not found for agency.",
-  "developerMessage": "(developer message)Locations groups not found for agency."
+  "userMessage": "(user message)Locations not found for agency not_an_agency with location type CELL",
```
Comment (matt-crowson-moj) [inline src/test/resources/__files/prisonapi/location-404.json:3]: Just a general question (and I know agency has different meanings).  Given out endpoints are looking up prisons then favour prison over agency in messages?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/ActivityScheduleInstanceControllerTest.kt [hunk 3]
```diff
@@ -156,7 +156,7 @@ class ActivityScheduleInstanceControllerTest : ControllerTestBase<ActivitySchedu
   }
 
   @Test
-  fun `200 response when date range equals 3 moths`() {
+  fun `200 response when date range equals 3 months`() {
```
Comment (timharrison-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/ActivityScheduleInstanceControllerTest.kt:159]: Haha. 3 moths!

**Review by timharrison-moj — APPROVED** [review]
LGTM (matt had a comment though)

**Review by matt-crowson-moj — APPROVED** [review]
LGTM

File: src/test/resources/__files/prisonapi/location-404.json (hunk 2 above)
Comment (markford-moj) [reply src/test/resources/__files/prisonapi/location-404.json:3]: Good question - although this is just test data - the messages are indicative of what the prison api would response with. Prison api often uses the term agency

**Pull request merged**